space-concordia-robotics/robotics-prototype
# Nir software ubuntu 513

# Assignee Section

## Description
Implements the DLP NIRScan Nano EVM software stuff so it can run on Linux. Note it does everything through USB commands and you can't use the GUI that TI provided you with cuz it only works on Windows. Note I had to manually transfer the files from Windows so there is no scripted installation or anything like that, the library is in-source  

Upon connecting the NIR through USB, once the main executable is run, it will scan whatever its pointing to and record the values in a CSV file, where the first column is a bunch of wavelengths and the second column are the absorbances for those wavelengths (Steven said he only needed absorbance) .

Since this just works on the terminal, the issue after this one will be to provide a very basic interface to our Web GUI with a few things : A start scan button, a little thing to create a scan configuration (it uses about 5-6 parameters), a button to select the current scan configuration, and a plot plotting the CSV file.

### Steps for Testing
- Go to `rover/science/NIR`
- `mkdir build`
- `cd build`
- `cmake ..`
- `make`
Since im the only one who has the NIR there really isnt any way for anyone else to test the actual device so i think just checking that the code builds should work. 

closes #[513]

# Reviewer Section
